### PR TITLE
Fixes for 10.0

### PR DESCRIPTION
**Pull request by MyGamesDevelopmentAcc**
Wasn't making any updates to spells, just modified it so it works without errors. Did some testing, haven't noticied any errors.

Changes:
Changes required by new handling of cast bar. Kept back compatiblity with color setting as much as I could best to my knowledge

### CODE REVIEW

**Comment by MyGamesDevelopmentAcc** [comment]
Hiding Blizzard frames does not yet work properly.

**Comment by MyGamesDevelopmentAcc** [comment]
They seemto hide now, but I think there is still an issue with setting focus by mouse

**Comment by MyGamesDevelopmentAcc** [comment]
The issue with focus and target is resolved.

**Comment by jturel** [comment]
@MyGamesDevelopmentAcc let me know when you don't expect to push any more changes and I can test

**Comment by MyGamesDevelopmentAcc** [comment]
I do not expect to push any more changes. To be honest I am not too fluent in githubish and didn't know initially that my 2nd commit through the github app will be attached automatically to this merge request, so I had to keep on rolling with it ;) Sorry about that.

**Comment by jturel** [comment]
> I do not expect to push any more changes. To be honest I am not too fluent in githubish and didn't know initially that my 2nd commit through the github app will be attached automatically to this merge request, so I had to keep on rolling with it ;) Sorry about that.

no worries. pushing to the branch that you created this PR from also updates the PR!

File: sArena.toc [hunk 1]
```diff
@@ -1,8 +1,8 @@
-## Interface: 90005
+## Interface: 100000
 ## Title: sArena
 ## Notes: Enhanced arena frames
 ## Author: Stako
-## Version: 3.1.8
+## Version: 3.1.9
```
Comment (jturel) [inline sArena.toc:5]: Make this 3.2 to signify a more significant change (backward incompatible with old version)

File: sArena.xml [hunk 2]
```diff
@@ -106,14 +107,16 @@
...
         </Layers>
-        <Scripts>
-            <OnLoad function="CastingBarFrame_OnLoad"/>
-            <OnEvent function="CastingBarFrame_OnEvent"/>
-            <OnUpdate function="CastingBarFrame_OnUpdate"/>
-            <OnShow function="CastingBarFrame_OnShow"/>
-        </Scripts>
         <BarTexture file="Interface\TargetingFrame\UI-StatusBar"/>
         <BarColor r="1.0" g="0.7" b="0.0"/>
+		<Scripts>
```
Comment (jturel) [inline sArena.xml:112]: Tab indentation

File: sArena.lua [hunk 3]
```diff
@@ -578,7 +580,7 @@ function sArenaFrameMixin:ResetTrinket()
 end
 
 local function ResetStatusBar(f)
-    f:SetStatusBarTexture(nil)
+    --f:SetStatusBarTexture(nil)
```
Comment (jturel) [inline sArena.lua:583]: What do you think about just deleting this line?

File: sArena.lua [hunk 4]
```diff
@@ -294,10 +294,12 @@ function sArenaFrameMixin:OnLoad()
     self:RegisterForClicks("AnyUp")
     self:SetAttribute("*type1", "target")
     self:SetAttribute("*type2", "focus")
+    self:SetAttribute("*typerelease1", "target") -- I do not know why this is treated as a typerelease, don't know how to remove it.
+    self:SetAttribute("*typerelease2", "focus")
     self:SetAttribute("unit", unit)
     self.unit = unit
 
-    CastingBarFrame_SetUnit(self.CastBar, unit, false, true)
+	self.CastBar:SetUnit(unit, false, true)
```
Comment (jturel) [inline sArena.lua:302]: Tab

File: sArena.lua [hunk 5]
```diff
@@ -63,7 +63,7 @@ local UnitFrameHealPredictionBars_Update = UnitFrameHealPredictionBars_Update
 local function UpdateBlizzVisibility(instanceType)
     -- hide blizz arena frames while in arena
     if ( InCombatLockdown() ) then return end
-    if ( not IsAddOnLoaded("Blizzard_ArenaUI") ) then return end
+    --if ( not IsAddOnLoaded("Blizzard_ArenaUI") ) then return end
```
Comment (jturel) [inline sArena.lua:66]: Delete this, or keep?

File: Layouts/Xaryu.lua [hunk 6]
```diff
@@ -229,6 +229,11 @@ end
 function layout:UpdateTextures(frame)
     local texture = self.db.classicBars and "Interface\\TargetingFrame\\UI-StatusBar" or "Interface\\RaidFrame\\Raid-Bar-Hp-Fill"
 
+	frame.CastBar.typeInfo = {
```
Comment (jturel) [inline Layouts/Xaryu.lua:232]: Tab indentation

File: Layouts/BlizzTourney.lua [hunk 7]
```diff
@@ -123,7 +123,13 @@ function layout:Initialize(frame)
     specBorder:Show()
 
     f = frame.CastBar
-    f:SetStatusBarTexture("Interface\\RaidFrame\\Raid-Bar-Hp-Fill")
+	local typeInfoTexture = "Interface\\RaidFrame\\Raid-Bar-Hp-Fill";
+    f:SetStatusBarTexture(typeInfoTexture)
+	f.typeInfo = {
```
Comment (jturel) [inline Layouts/BlizzTourney.lua:128]: Tab indentation

File: Layouts/BlizzTarget.lua [hunk 8]
```diff
@@ -128,7 +128,13 @@ function layout:Initialize(frame)
     f:SetHeight(12)
 
     f = frame.CastBar
-    f:SetStatusBarTexture("Interface\\TargetingFrame\\UI-StatusBar")
+	local typeInfoTexture = "Interface\\TargetingFrame\\UI-StatusBar";
+    f:SetStatusBarTexture(typeInfoTexture)
+	f.typeInfo = {
```
Comment (jturel) [inline Layouts/BlizzTarget.lua:133]: Tab indentation

File: Layouts/BlizzArena.lua [hunk 9]
```diff
@@ -127,7 +127,13 @@ function layout:Initialize(frame)
     f:SetHeight(12)
 
     f = frame.CastBar
-    f:SetStatusBarTexture("Interface\\TargetingFrame\\UI-StatusBar")
+	local typeInfoTexture = "Interface\\TargetingFrame\\UI-StatusBar";
```
Comment (jturel) [inline Layouts/BlizzArena.lua:130]: Tab indentation

**Review by jturel — CHANGES_REQUESTED** [review]
Thank you @MyGamesDevelopmentAcc - this is a big improvement!

I left a few comments - mostly around using tab indentation vs space indentation. Better to be consistent with what's already there :)

**Comment by MyGamesDevelopmentAcc** [comment]
@jturel I have made the requested fixes, added some further formatting at the end, fixed further problems found with WoW10 handling of SecureActionButtons in certain setups.

**Comment by jturel** [comment]
> I am also thinking that this will most likely break, if there was, the compatiblity with older Wow versions as I have not added any special handling to distinguish whether this is wow10 or not.

As far as I'm aware the "retail" version of sArena has supported only the latest wow version!

**Comment by jturel** [comment]
This got a lot harder to review because of all the extra formatting changes that got included. I'll need some time :)

**Comment by MyGamesDevelopmentAcc** [comment]
For that reason I have done it in separate commits, hoping it will be easier to see which was changed/modified in each.

**Comment by jturel** [comment]
@MyGamesDevelopmentAcc i fixed an automatic action that should run in this repo to validate the lua syntax automatically. Can you do me a favor and rebase your branch against the master branch? After you push, you'll see the job run here and tell you whether it passes or fails.

Your steps would look something like the below, given a remote called 'origin' corresponding to this repo (not your fork):

```
git fetch origin
git rebase origin/master
git push $YOUR_FORK_REMOTE $YOUR_BRANCH_NAME --force
```

**Comment by MyGamesDevelopmentAcc** [comment]
Added two new globals and rebased. Please let me know if this is what you asked for.

File: .luacheckrc [hunk 10]
```diff
@@ -49,5 +49,7 @@ read_globals = {
     'UnitPowerMax',
     'UnitPowerType',
     'UnitRace',
+    'ArenaEnemyFramesContainer',
```
Comment (jturel) [inline .luacheckrc:52]: This looks good. Can you put these at the top of the file? They seem to be in alphabetical order :)

**Comment by jturel** [comment]
Thank you @MyGamesDevelopmentAcc ! Unfortunately this is as far as I can help you as I don't have the ability to create a new release on Curse. Merging this.

**Review by jturel — APPROVED** [review]

**Comment by MyGamesDevelopmentAcc** [comment]
@stako any chances for you to release? @jturel maybe we can add release format which could be picked by wowup?

**Comment by jturel** [comment]
> @stako any chances for you to release this please? @jturel maybe we can add release format which could be picked by wowup?

hey there @MyGamesDevelopmentAcc - missed this message. I'm happy to review and test any changes to get the addon picked up by wowup :)

**Pull request merged**